File: sdk/src/main/java/com/qonversion/android/sdk/logger/ConsoleLogger.java
package com.qonversion.android.sdk.logger;

import android.util.Log;

import com.qonversion.android.sdk.BuildConfig;

public class ConsoleLogger implements Logger {

    private static final String TAG = "Qonversion";

    @Override
    public void release(String message) {
        log(TAG, message);
    }

    @Override
    public void debug(String message) {
        if (BuildConfig.DEBUG) {
            log(TAG, message);
        }
    }

    @Override
    public void debug(String tag, String message) {
        if (BuildConfig.DEBUG) {
            log(tag, message);
        }
    }

    private void log(String tag, String message){
        Log.println(Log.DEBUG, tag, format(message));
    }

    private String format(final String message) {
        return "Thread - " + Thread.currentThread().getName() + " " + message;
    }
}

File: sdk/src/main/java/com/qonversion/android/sdk/internal/logger/ConsoleLogger.kt
package com.qonversion.android.sdk.internal.logger

import android.util.Log
import com.qonversion.android.sdk.BuildConfig

internal class ConsoleLogger : Logger {
    override fun error(message: String) {
        log(Log.ERROR, message)
    }

    override fun warn(message: String) {
        log(Log.WARN, message)
    }

    override fun release(message: String) {
        log(Log.INFO, message)
    }

    override fun debug(message: String) {
        if (BuildConfig.DEBUG) {
            log(Log.DEBUG, message)
        }
    }

    private fun log(logLevel: Int, message: String) {
        Log.println(logLevel, TAG, format(message))
    }

    private fun format(message: String): String {
        return "[Thread - " + Thread.currentThread().name + "] " + message
    }

    companion object {
        private const val TAG = "Qonversion"
    }
}
